src/sentry/api/permissions.py

```diff
@@ -45,16 +45,40 @@ def has_permission(self, request: Request, view: object) -> bool:
 
 
 class SuperuserPermission(BasePermission):
+    """
+    This permission class is used for endpoints that should ONLY be accessible
+    by superuser.
+    """
+
     def has_permission(self, request: Request, view: object) -> bool:
         return is_active_superuser(request)
 
 
 class StaffPermission(BasePermission):
+    """
+    This permission class is used for endpoints that should ONLY be accessible
+    by staff.
+    """
+
     def has_permission(self, request: Request, view: object) -> bool:
         return is_active_staff(request)
 
 
-# XXX(schew2381): This is a temporary permission that does NOT perform an OR
+class StaffPermissionMixin:
+    """
+    Sentry endpoints that should be accessible by staff but have an existing permission
+    class (that is not StaffPermission) require this mixin because staff does not give
+    any scopes. See 'OrganizationAndStaffPermission' for an example of this.
+    """
+
+    def has_permission(self, request, *args, **kwargs):
+        return super().has_permission(request, *args, **kwargs) or is_active_staff(request)
+
+    def has_object_permission(self, request, *args, **kwargs):
```

Comment:
what is an object permission vs. a regular permission?

Reply:
We call `has_permission` as a base check when your request comes in. Most of our permission classes don't override this. `has_object_permission` is more specific to the type of request you're making (e.g. team or org request) as is called from `convert_args` most of the time